### ISSUE TITLE
build number displayed on home screen is stuck at (1)

### ISSUE THREAD

**Issue by scottleibrand** [Co-Epi/app-android#110]
I'm using the version tagged as Android 0.2 build 83 in MS App Center, and on the app home screen it reports its version as 0.2 (1).

**Comment by yibter** [comment]
Looks like @i-schuetz just updated the name & code version values to 0.2.1 (2) via this commit..

https://github.com/Co-Epi/app-android/commit/e64a035f5b7a154733ef09db0d8d7c0d5cb5efa2

I've verified the running application reflects this change. Does that resolve this ticket @scottleibrand? Or are you wanting the application to use a different build number generated by MS App Center when a build is generated there?

If it's the latter, maybe someone with MS App Center access/experience might want to review this ticket. Looks like there is an "increment version number" feature that can be enabled, but sounds like it may only update the versionCode in AndroidManifest.xml...

https://docs.microsoft.com/en-us/appcenter/build/android/first-build#34-increment-version-number

However, we currently we have the versionCode defined in build.gradle and it will override any AndroidManifest.xml value. So we can relocate the versionCode to AndroidManifest.xml or another option might be to reference an environment variable in build.gradle...

https://docs.microsoft.com/en-us/appcenter/build/custom/variables/#buildgradle-for-android

Or a pre-build step to update build.gradle...
https://docs.microsoft.com/en-us/appcenter/build/custom/scripts/#pre-build

Thoughts?

**Comment by yibter** [comment]
Just created a PR to relocate versionCode from build.gradle to AndroidManifest.xml so the MS App Center "increment version number" feature will work if someone wants to try it out...

https://github.com/Co-Epi/app-android/pull/130

**Comment by mellowgeek** [comment]
@yibter can you move this branch onto the main repo instead of your fork so I can try building it in appcenter?

**Comment by yibter** [comment]
> @yibter can you move this branch onto the main repo instead of your fork so I can try building it in appcenter?

Can do @mellowgeek! Though, I was able to spin up the app using personal instance of appcenter with the change and it's still not show the appcenter's build number on the UI. I've downloaded & verified the appcenter built APK has the generated versionCode in the AndroidManifest.xml... and I've verified building locally that the UI will show the AndroidManifest.xml versionCode value. But yeah, the VERSION_CODE value created in BuildConfig via appcenter built APK is using the value hardcoded in the branch and not the one it is generating and putting in the AndroidManifest.xml. So yeah, still reviewing what the cleanest/recommended solution is for appcenter and then I'll do another PR once I verify it's working.
